### PR TITLE
Fix ID saving

### PR DESCRIPTION
**Pull request by MrIbby**

### CODE REVIEW

**Comment by MrIbby** [comment]
Block states will be in another PR.

File: src/main/java/xyz/openmodloader/util/TriConsumer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    /**
+     * @param after the operation to perform after this operation
+     * @return a composed {@code TriConsumer} that performs in sequence this
+     * operation followed by the {@code after} operation
+     * @throws NullPointerException if {@code after} is null
+     */
+    default TriConsumer<T, U, V> andThen(TriConsumer<? super T, ? super U, ? super V> after) {
+        Objects.requireNonNull(after);
+
```
Comment (phase) [inline src/main/java/xyz/openmodloader/util/TriConsumer.java]: this newline doesn't seem required

File: src/main/java/xyz/openmodloader/world/storage/OMLSaveHandler.java [hunk 2]
```diff
@@ -25,26 +25,21 @@
...
-//                registeredObject.setInteger("value", registry.getIDForObject(registry.getObject(key)));
-//                registeredObjectList.appendTag(registeredObject);
-//            }
-//            registryNBT.setTag("Data", registeredObjectList);
-//
-//            registryList.appendTag(registryNBT);
-//        });
-//
-//        nbt.setTag("Registry", registryList);
+        NBTTagList registryList = new NBTTagList();
```
Comment (phase) [inline src/main/java/xyz/openmodloader/world/storage/OMLSaveHandler.java]: The new lines that were removed seemed fine. This code is too close together.

**Comment by MrIbby** [comment]
Anything else?

File: src/main/java/xyz/openmodloader/registry/AutomaticNamespacedRegistry.java [hunk 3]
```diff
@@ -29,46 +32,56 @@ public AutomaticNamespacedRegistry(Function<String, K> keyFactory) {
...
+        registryObjects.forEach(action);
+    }
+
+    /**
+     * Performs the given action for each object in this registry.
+     *
+     * @param action the action to be performed for each object
+     * @throws NullPointerException if the specified action is null
+     */
+    public void forEach(TriConsumer<Integer, K, V> action) {
```
Comment (coolsquid) [inline src/main/java/xyz/openmodloader/registry/AutomaticNamespacedRegistry.java]: Is a TriConsumer really necessary? Can't you just pass the key and the value, and let people call getId() to get the ID?

Comment (MrIbby) [inline src/main/java/xyz/openmodloader/registry/AutomaticNamespacedRegistry.java]: Does it matter? It doesn't hurt anyone and makes it easier to iterate over the registry.

**Comment by MrIbby** [comment]
🎉

**Pull request merged**